src/main/kotlin/nl/avisi/structurizr/site/generatr/ClonedRepository.kt

```diff
@@ -48,4 +50,17 @@ class ClonedRepository(
             .setName(branch)
             .call()
     }
+
+    fun getBranchNames(excludeBranches: List<String>): List<String> {
+        val branches:List<Ref> = Git(repo).branchList().setListMode(ListBranchCommand.ListMode.ALL).call()
+        val branchNames:MutableList<String> = mutableListOf()
+        for (branch:Ref in branches){
+            val shortBranchName: String = branch.name.toString().substringAfterLast("/")
+            println("Found the following Branch $shortBranchName")
+            if(!excludeBranches.contains(shortBranchName)){
+                branchNames.add(shortBranchName)
+            }
+        }
+        return branchNames
+    }
```

Comment:
Working with mutable lists like this is not really idiomatic Kotlin. This is much more concise and more idiomatic:
```suggestion
    fun getBranchNames(excludeBranches: List<String>) =
        Git(repo).branchList().setListMode(ListBranchCommand.ListMode.ALL).call()
            .map { it.name.toString().substringAfterLast("/") }
            .onEach { println("Found the following branch: $it") }
            .filter { it !in excludeBranches }
```